<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />


<meta name="author" content="Stefano Cacciatore" />

<meta name="date" content="2024-09-18" />

<title>Statistics</title>

<script src="site_libs/header-attrs-2.27/header-attrs.js"></script>
<script src="site_libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/united.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<style>h1 {font-size: 34px;}
       h1.title {font-size: 38px;}
       h2 {font-size: 30px;}
       h3 {font-size: 24px;}
       h4 {font-size: 18px;}
       h5 {font-size: 16px;}
       h6 {font-size: 12px;}
       code {color: inherit; background-color: rgba(0, 0, 0, 0.04);}
       pre:not([class]) { background-color: white }</style>
<script src="site_libs/jqueryui-1.13.2/jquery-ui.min.js"></script>
<link href="site_libs/tocify-1.9.1/jquery.tocify.css" rel="stylesheet" />
<script src="site_libs/tocify-1.9.1/jquery.tocify.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<link href="site_libs/highlightjs-9.12.0/textmate.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>
<link href="site_libs/font-awesome-6.4.2/css/all.min.css" rel="stylesheet" />
<link href="site_libs/font-awesome-6.4.2/css/v4-shims.min.css" rel="stylesheet" />

<link rel="icon" href="https://github.com/workflowr/workflowr-assets/raw/main/img/reproducible.png">
<!-- Add a small amount of space between sections. -->
<style type="text/css">
div.section {
  padding-top: 12px;
}
</style>



<style type="text/css">
  code{white-space: pre-wrap;}
  span.smallcaps{font-variant: small-caps;}
  span.underline{text-decoration: underline;}
  div.column{display: inline-block; vertical-align: top; width: 50%;}
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
  ul.task-list{list-style: none;}
    </style>

<style type="text/css">code{white-space: pre;}</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>









<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
img {
  max-width:100%;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
details > summary > p:only-child {
  display: inline;
}
pre code {
  padding: 0;
}
</style>


<style type="text/css">
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #adb5bd;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script type="text/javascript">
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark the anchor link active (and if it's in a dropdown, also mark that active)
  var dropdown = menuAnchor.closest('li.dropdown');
  if (window.bootstrap) { // Bootstrap 4+
    menuAnchor.addClass('active');
    dropdown.find('> .dropdown-toggle').addClass('active');
  } else { // Bootstrap 3
    menuAnchor.parent().addClass('active');
    dropdown.addClass('active');
  }

  // Navbar adjustments
  var navHeight = $(".navbar").first().height() + 15;
  var style = document.createElement('style');
  var pt = "padding-top: " + navHeight + "px; ";
  var mt = "margin-top: -" + navHeight + "px; ";
  var css = "";
  // offset scroll position for anchor links (for fixed navbar)
  for (var i = 1; i <= 6; i++) {
    css += ".section h" + i + "{ " + pt + mt + "}\n";
  }
  style.innerHTML = "body {" + pt + "padding-bottom: 40px; }\n" + css;
  document.head.appendChild(style);
});
</script>

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before, .tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "\e259";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "\e258";
  font-family: 'Glyphicons Halflings';
  border: none;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
  background-color: transparent;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<!-- code folding -->



<style type="text/css">

#TOC {
  margin: 25px 0px 20px 0px;
}
@media (max-width: 768px) {
#TOC {
  position: relative;
  width: 100%;
}
}

@media print {
.toc-content {
  /* see https://github.com/w3c/csswg-drafts/issues/4434 */
  float: right;
}
}

.toc-content {
  padding-left: 30px;
  padding-right: 40px;
}

div.main-container {
  max-width: 1200px;
}

div.tocify {
  width: 20%;
  max-width: 260px;
  max-height: 85%;
}

@media (min-width: 768px) and (max-width: 991px) {
  div.tocify {
    width: 25%;
  }
}

@media (max-width: 767px) {
  div.tocify {
    width: 100%;
    max-width: none;
  }
}

.tocify ul, .tocify li {
  line-height: 20px;
}

.tocify-subheader .tocify-item {
  font-size: 0.90em;
}

.tocify .list-group-item {
  border-radius: 0px;
}


</style>



</head>

<body>


<div class="container-fluid main-container">


<!-- setup 3col/9col grid for toc_float and main content  -->
<div class="row">
<div class="col-xs-12 col-sm-4 col-md-3">
<div id="TOC" class="tocify">
</div>
</div>

<div class="toc-content col-xs-12 col-sm-8 col-md-9">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-bs-toggle="collapse" data-target="#navbar" data-bs-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">R Tutorial</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">Home</a>
</li>
<li>
  <a href="Install.html">R install</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">
    Basic
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
    <li>
      <a href="Introduction.html">Introduction to R and RStudio</a>
    </li>
    <li>
      <a href="Statistics.html">Statistics</a>
    </li>
    <li>
      <a href="Visualisation.html">Data Visualisation</a>
    </li>
  </ul>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">
    Multivariate analysis
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
    <li>
      <a href="Supervised_learning.html">Supervised Learning</a>
    </li>
    <li>
      <a href="Unsupervised_Learning.html">Unsupervised Learning</a>
    </li>
  </ul>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">
    Applications
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
    <li>
      <a href="TCGA.html">TCGA</a>
    </li>
    <li>
      <a href="Single-cell.html">Single-cell</a>
    </li>
  </ul>
</li>
<li>
  <a href="license.html">License</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        <li>
  <a href="https://github.com/oliverdesousa/R_Tutorials">
    <span class="fab fa-github"></span>
     
    Source code
  </a>
</li>
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div id="header">



<h1 class="title toc-ignore">Statistics</h1>
<h4 class="author">Stefano Cacciatore</h4>
<h4 class="date">September 18, 2024</h4>

</div>


<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-report" data-toggle="collapse" data-target="#workflowr-report">
<span class="glyphicon glyphicon-list" aria-hidden="true"></span>
workflowr <span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span>
</button>
</p>
<div id="workflowr-report" class="collapse">
<ul class="nav nav-tabs">
<li class="active">
<a data-toggle="tab" href="#summary">Summary</a>
</li>
<li>
<a data-toggle="tab" href="#checks"> Checks <span
class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span>
</a>
</li>
<li>
<a data-toggle="tab" href="#versions">Past versions</a>
</li>
</ul>
<div class="tab-content">
<div id="summary" class="tab-pane fade in active">
<p>
<strong>Last updated:</strong> 2024-09-18
</p>
<p>
<strong>Checks:</strong> <span
class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> 7
<span class="glyphicon glyphicon-exclamation-sign text-danger"
aria-hidden="true"></span> 0
</p>
<p>
<strong>Knit directory:</strong> <code>Tutorials/</code> <span
class="glyphicon glyphicon-question-sign" aria-hidden="true"
title="This is the local directory in which the code in this file was executed.">
</span>
</p>
<p>
This reproducible <a href="https://rmarkdown.rstudio.com">R Markdown</a>
analysis was created with <a
  href="https://github.com/workflowr/workflowr">workflowr</a> (version
1.7.1). The <em>Checks</em> tab describes the reproducibility checks
that were applied when the results were created. The <em>Past
versions</em> tab lists the development history.
</p>
<hr>
</div>
<div id="checks" class="tab-pane fade">
<div id="workflowr-checks" class="panel-group">
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRMarkdownfilestronguptodate">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>R Markdown file:</strong> up-to-date
</a>
</p>
</div>
<div id="strongRMarkdownfilestronguptodate"
class="panel-collapse collapse">
<div class="panel-body">
<p>Great! Since the R Markdown file has been committed to the Git
repository, you know the exact version of the code that produced these
results.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongEnvironmentstrongempty">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Environment:</strong> empty </a>
</p>
</div>
<div id="strongEnvironmentstrongempty" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! The global environment was empty. Objects defined in the
global environment can affect the analysis in your R Markdown file in
unknown ways. For reproduciblity it’s best to always run the code in an
empty environment.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSeedstrongcodesetseed20240905code">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Seed:</strong>
<code>set.seed(20240905)</code> </a>
</p>
</div>
<div id="strongSeedstrongcodesetseed20240905code"
class="panel-collapse collapse">
<div class="panel-body">
<p>The command <code>set.seed(20240905)</code> was run prior to running
the code in the R Markdown file. Setting a seed ensures that any results
that rely on randomness, e.g. subsampling or permutations, are
reproducible.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSessioninformationstrongrecorded">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Session information:</strong>
recorded </a>
</p>
</div>
<div id="strongSessioninformationstrongrecorded"
class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Recording the operating system, R version, and package
versions is critical for reproducibility.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongCachestrongnone">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Cache:</strong> none </a>
</p>
</div>
<div id="strongCachestrongnone" class="panel-collapse collapse">
<div class="panel-body">
<p>Nice! There were no cached chunks for this analysis, so you can be
confident that you successfully produced the results during this
run.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongFilepathsstrongrelative">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>File paths:</strong> relative </a>
</p>
</div>
<div id="strongFilepathsstrongrelative" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Using relative paths to the files within your workflowr
project makes it easier to run your code on other machines.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRepositoryversionstrongahrefhttpsgithubcomtkcacciaTutorialstreecbb2736ddeb6de44f10e495f42b08d3b86d14faftargetblankcbb2736a">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Repository version:</strong>
<a href="https://github.com/tkcaccia/Tutorials/tree/cbb2736ddeb6de44f10e495f42b08d3b86d14faf" target="_blank">cbb2736</a>
</a>
</p>
</div>
<div
id="strongRepositoryversionstrongahrefhttpsgithubcomtkcacciaTutorialstreecbb2736ddeb6de44f10e495f42b08d3b86d14faftargetblankcbb2736a"
class="panel-collapse collapse">
<div class="panel-body">
<p>
Great! You are using Git for version control. Tracking code development
and connecting the code version to the results is critical for
reproducibility.
</p>
<p>
The results in this page were generated with repository version
<a href="https://github.com/tkcaccia/Tutorials/tree/cbb2736ddeb6de44f10e495f42b08d3b86d14faf" target="_blank">cbb2736</a>.
See the <em>Past versions</em> tab to see a history of the changes made
to the R Markdown and HTML files.
</p>
<p>
Note that you need to be careful to ensure that all relevant files for
the analysis have been committed to Git prior to generating the results
(you can use <code>wflow_publish</code> or
<code>wflow_git_commit</code>). workflowr only checks the R Markdown
file, but you know if there are other scripts or data files that it
depends on. Below is the status of the Git repository when the results
were generated:
</p>
<pre><code>
Ignored files:
    Ignored:    .DS_Store
    Ignored:    data/.DS_Store

Unstaged changes:
    Deleted:    data/COADREAD.clin.merged.picked.txt
    Deleted:    data/COADREAD.rnaseqv2__illuminahiseq_rnaseqv2__unc_edu__Level_3__RSEM_genes_normalized__data.data.txt

</code></pre>
<p>
Note that any generated files, e.g. HTML, png, CSS, etc., are not
included in this status report because it is ok for generated content to
have uncommitted changes.
</p>
</div>
</div>
</div>
</div>
<hr>
</div>
<div id="versions" class="tab-pane fade">

<p>
These are the previous versions of the repository in which changes were
made to the R Markdown (<code>analysis/Statistics.Rmd</code>) and HTML
(<code>docs/Statistics.html</code>) files. If you’ve configured a remote
Git repository (see <code>?wflow_git_remote</code>), click on the
hyperlinks in the table below to view the files as they were in that
past version.
</p>
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
File
</th>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
<th>
Message
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/tkcaccia/Tutorials/83f8d4e746d713066af6582f8ceaa6cccb31b6d1/docs/Statistics.html" target="_blank">83f8d4e</a>
</td>
<td>
tkcaccia
</td>
<td>
2024-09-16
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/tkcaccia/Tutorials/159190a0f38e56bd72f396b250277a4a3b3b74bf/docs/Statistics.html" target="_blank">159190a</a>
</td>
<td>
tkcaccia
</td>
<td>
2024-09-16
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/tkcaccia/Tutorials/blob/dff6a04355ffd83b249f5c019a422bbc66a22717/analysis/Statistics.Rmd" target="_blank">dff6a04</a>
</td>
<td>
tkcaccia
</td>
<td>
2024-09-16
</td>
<td>
Start my new project
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/tkcaccia/Tutorials/6d23cdb74595a082fc1d86dd7d29bc5ce65703f1/docs/Statistics.html" target="_blank">6d23cdb</a>
</td>
<td>
tkcaccia
</td>
<td>
2024-09-16
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/tkcaccia/Tutorials/blob/8b0b07aea50228ab49a3660e7b1327cb440a37c1/analysis/Statistics.Rmd" target="_blank">8b0b07a</a>
</td>
<td>
tkcaccia
</td>
<td>
2024-09-16
</td>
<td>
Start my new project
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/tkcaccia/Tutorials/6301d0ade76fe73d7038cd24aecc9f9f9df2794e/docs/Statistics.html" target="_blank">6301d0a</a>
</td>
<td>
tkcaccia
</td>
<td>
2024-09-16
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/tkcaccia/Tutorials/blob/0119c71fa040a7bbc8c03690458109f12a47b9ac/analysis/Statistics.Rmd" target="_blank">0119c71</a>
</td>
<td>
tkcaccia
</td>
<td>
2024-09-16
</td>
<td>
Start my new project
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/tkcaccia/Tutorials/897778af75990c4268c8b31f4717b5182769c81b/docs/Statistics.html" target="_blank">897778a</a>
</td>
<td>
tkcaccia
</td>
<td>
2024-09-16
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/tkcaccia/Tutorials/1a34734cb3ef1db7012e9458c9d98fcd6625835a/docs/Statistics.html" target="_blank">1a34734</a>
</td>
<td>
tkcaccia
</td>
<td>
2024-09-16
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/tkcaccia/Tutorials/5033c128d8bca78746899004bd82e65b109bffd4/docs/Statistics.html" target="_blank">5033c12</a>
</td>
<td>
tkcaccia
</td>
<td>
2024-09-16
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/tkcaccia/Tutorials/blob/e9cf75197fecd2b17c49839ac8157b084819504e/analysis/Statistics.Rmd" target="_blank">e9cf751</a>
</td>
<td>
tkcaccia
</td>
<td>
2024-09-16
</td>
<td>
Start my new project
</td>
</tr>
</tbody>
</table>
</div>
<hr>
</div>
</div>
</div>
<div id="univariate-analysis" class="section level2">
<h2>Univariate Analysis</h2>
<p><em>What is univariate analysis ?</em></p>
<p>The idea of univariate analysis is to first understand the variables
individually. It is typically the first step in understanding a data
set. A variable in UA is a condition or subset that your data falls
into. You can think of it as a “category” such as “age”, “weight” or
“length”. However, UA does not look at &gt; than 1 variable at a time
(this would be a bivariate analysis)</p>
<div id="learning-objectives" class="section level4">
<h4>Learning Objectives:</h4>
<ul>
<li><p>Summarising Data</p></li>
<li><p>Frequency Tables</p></li>
<li><p>Univariate Hypothesis Testing</p></li>
<li><p>Visualising Univariate Data</p></li>
<li><p>Correlation</p></li>
<li><p>Simple Regression analysis</p></li>
</ul>
<pre class="r"><code># Installation of packages (usually needed)
# install.packages(&quot;ggplot2&quot;)
# install.packages(&quot;dplyr&quot;)
# install.packages(&quot;ggpubr&quot;)
# install.packages(&quot;corrplot&quot;)

# Loading of packages
library(ggplot2)
library(dplyr)
library(ggpubr)
library(corrplot)
library(stats)</code></pre>
</div>
<div id="summarising-data" class="section level3">
<h3>1. Summarising Data</h3>
<pre class="r"><code># Using the data set stored in Rstudio called &quot;cars&quot;

# We need to create an array of our single variable for UA:
x &lt;- cars$speed</code></pre>
<p>Looking at the <strong>CENTRAL TENDENCY</strong> of the data:</p>
<pre class="r"><code>mean(x)</code></pre>
<pre><code>[1] 15.4</code></pre>
<pre class="r"><code>median(x)</code></pre>
<pre><code>[1] 15</code></pre>
<pre class="r"><code>mode(x)</code></pre>
<pre><code>[1] &quot;numeric&quot;</code></pre>
<p>Looking at the <strong>DISPERSION</strong> of the data:</p>
<pre class="r"><code>min(x)</code></pre>
<pre><code>[1] 4</code></pre>
<pre class="r"><code>max(x)</code></pre>
<pre><code>[1] 25</code></pre>
<pre class="r"><code># Range of the data:
range(x)</code></pre>
<pre><code>[1]  4 25</code></pre>
<pre class="r"><code># Inter-quantile range:
IQR(x)</code></pre>
<pre><code>[1] 7</code></pre>
<pre class="r"><code># Variance --&gt;
var(x)</code></pre>
<pre><code>[1] 27.95918</code></pre>
<pre class="r"><code># Standard Deviation:
sd(x)</code></pre>
<pre><code>[1] 5.287644</code></pre>
<p><strong><em>TIP:</em></strong> you can use the function
<code>summary</code> to produce result summaries of the results of
various model fitting functions.</p>
<pre class="r"><code>summary(x)</code></pre>
<pre><code>   Min. 1st Qu.  Median    Mean 3rd Qu.    Max. 
    4.0    12.0    15.0    15.4    19.0    25.0 </code></pre>
</div>
<div id="frequency-tables" class="section level3">
<h3>2. Frequency Tables:</h3>
<ul>
<li><p>The frequency of an observation tells you the number of times the
observation occurs in the data.</p></li>
<li><p>A frequency table is a collection of these observations and their
frequencies.</p></li>
<li><p>A frequency table can be shown either graphically (bar
chart/histogram) or as a frequency distribution table.</p></li>
<li><p>These tables can show qualitative (categorical) or quantitative
(numeric) variables.</p></li>
</ul>
<div id="example-data" class="section level4">
<h4>Example Data</h4>
<p>We will use a data frame with a categorical variable and a numerical
variable to demonstrate each type of table.</p>
<pre class="r"><code># Create example data
set.seed(123)  # For reproducibility
data &lt;- data.frame(
  category = sample(c(&quot;A&quot;, &quot;B&quot;, &quot;C&quot;, &quot;D&quot;), 100, replace = TRUE),
  value = rnorm(100, mean = 50, sd = 10)
)
head(data)</code></pre>
<pre><code>  category    value
1        C 52.53319
2        C 49.71453
3        C 49.57130
4        B 63.68602
5        C 47.74229
6        B 65.16471</code></pre>
<pre class="r"><code># Frequency table for the categorical variable
freq_table &lt;- table(data$category)
freq_table</code></pre>
<pre><code>
 A  B  C  D 
28 26 29 17 </code></pre>
<p><strong>Note:</strong> the frequency table is CASE-SENSITIVE so the
frequencies of the variables corresponds to how many times that specific
number of string appears.</p>
</div>
<div id="grouped-tables" class="section level4">
<h4>Grouped Tables:</h4>
<p>Grouped tables aggregate the data into groups or bins.</p>
<pre class="r"><code># 1st Step: Create BINS for the numerical data
bins &lt;- cut(x, breaks = 5)
freq_table_numeric &lt;- table(bins)
freq_table_numeric</code></pre>
<pre><code>bins
 (3.98,8.2]  (8.2,12.4] (12.4,16.6] (16.6,20.8]   (20.8,25] 
          5          10          13          15           7 </code></pre>
<pre class="r"><code># Group data into bins and create a grouped table:
grouped_table &lt;- table(cut(x, breaks = 5))
grouped_table</code></pre>
<pre><code>
 (3.98,8.2]  (8.2,12.4] (12.4,16.6] (16.6,20.8]   (20.8,25] 
          5          10          13          15           7 </code></pre>
</div>
<div id="percentage-proportion-tables" class="section level4">
<h4>Percentage (Proportion) Tables</h4>
<p>Percentage tables show the proportion of each unique value or group
in the data.</p>
<pre class="r"><code># Percentage table for the categorical variable
percentage_table &lt;- prop.table(table(x)) * 100
percentage_table</code></pre>
<pre><code>x
 4  7  8  9 10 11 12 13 14 15 16 17 18 19 20 22 23 24 25 
 4  4  2  2  6  4  8  8  8  6  4  6  8  6 10  2  2  8  2 </code></pre>
<pre class="r"><code># Percentage table for the grouped numerical data
percentage_table_numeric &lt;- prop.table(table(cut(x, breaks = 5))) * 100
percentage_table_numeric</code></pre>
<pre><code>
 (3.98,8.2]  (8.2,12.4] (12.4,16.6] (16.6,20.8]   (20.8,25] 
         10          20          26          30          14 </code></pre>
</div>
<div id="cumulative-proportion-tables" class="section level4">
<h4>Cumulative Proportion Tables</h4>
<p>Cumulative proportion tables show the cumulative proportion of each
unique value or group.</p>
<pre class="r"><code># Cumulative proportion table for the categorical variable
cumulative_prop &lt;- cumsum(prop.table(table(data$category)))
cumulative_prop &lt;- cumulative_prop * 100
cumulative_prop</code></pre>
<pre><code>  A   B   C   D 
 28  54  83 100 </code></pre>
<pre class="r"><code># Cumulative proportion table for the grouped numerical data
cumulative_prop_numeric &lt;- cumsum(prop.table(table(cut(x, breaks = 5))))
cumulative_prop_numeric &lt;- cumulative_prop_numeric * 100
cumulative_prop_numeric</code></pre>
<pre><code> (3.98,8.2]  (8.2,12.4] (12.4,16.6] (16.6,20.8]   (20.8,25] 
         10          30          56          86         100 </code></pre>
<p><span style="color:red;"><strong><em>Question
1:</em></strong></span></p>
<p>Using the <code>cars</code> datset:</p>
<ol style="list-style-type: lower-alpha">
<li><p>Calculate the mean, median, and standard deviation of variable
“speed”.</p></li>
<li><p>Interpret what these statistics tell you about the speed
data.</p></li>
<li><p>Compute the range and interquartile range (IQR) of
speed.</p></li>
<li><p>What do these measures reveal about the dispersion of the speed
data?</p></li>
<li><p>Use the summary function to get a summary of x.</p></li>
<li><p>Describe the central tendency and dispersion metrics provided by
the summary output.</p></li>
</ol>
<p><span style="color:red;"><strong><em>Question
2:</em></strong></span></p>
<p><em>Using the below:</em></p>
<pre class="r"><code>xy &lt;- data.frame(
  category = sample(c(&quot;A&quot;, &quot;B&quot;, &quot;C&quot;, &quot;D&quot;), 100, replace = TRUE)
)
head(xy)</code></pre>
<pre><code>  category
1        B
2        B
3        A
4        B
5        A
6        A</code></pre>
<ol style="list-style-type: lower-alpha">
<li><p>Create a frequency table for the category variable.</p></li>
<li><p>What is the frequency of each category?</p></li>
</ol>
<p><em>Using the below:</em></p>
<pre class="r"><code>data &lt;- data.frame(
  value = rnorm(100, mean = 50, sd = 10)
)</code></pre>
<ol start="3" style="list-style-type: lower-alpha">
<li><p>Create a frequency table for the value variable.</p></li>
<li><p>How many observations fall into each unique value?</p></li>
</ol>
<p><em>Using the below:</em></p>
<pre class="r"><code>x &lt;- data$value
bins &lt;- cut(x, breaks = 5)</code></pre>
<ol start="5" style="list-style-type: lower-alpha">
<li><p>Create a grouped frequency table for the value variable using 5
bins.</p></li>
<li><p>What are the frequencies for each bin?</p></li>
</ol>
<p><em>Using the below:</em></p>
<pre class="r"><code>x &lt;- data$value
bins &lt;- cut(x, breaks = 5)</code></pre>
<ol start="7" style="list-style-type: lower-alpha">
<li><p>Create a percentage (proportion) table for the grouped value
data.</p></li>
<li><p>What percentage of the observations fall into each bin?</p></li>
</ol>
<details>
<summary>
Answers:
</summary>
<pre class="r"><code># Question 1:
# a. Calculate the mean, median, and standard deviation of variable &quot;speed&quot;
mean_speed &lt;- mean(x)
median_speed &lt;- median(x)
sd_speed &lt;- sd(x)

# c. Compute the range and interquartile range (IQR) of speed
range_speed &lt;- range(x)
iqr_speed &lt;- IQR(x)

# e. Use the summary function to get a summary of x
summary_speed &lt;- summary(x)

# Question 2:
# a. Create a frequency table for the category variable
freq_table_category &lt;- table(xy$category)

# c. Create a frequency table for the value variable
freq_table_value &lt;- table(data$value)

# e. Create a grouped frequency table for the value variable using 5 bins
grouped_table &lt;- table(bins)

# g. Create a percentage (proportion) table for the grouped value data
percentage_table &lt;- prop.table(grouped_table) * 100</code></pre>
</div>
</div>
<div id="univariate-hypothesis-testing" class="section level3">
<h3>3. Univariate Hypothesis Testing:</h3>
<p>Often, the data you are dealing with is a subset (sample) of the
complete data (population). Thus, the common question here is:</p>
<ul>
<li><em>Can the findings of the sample be extrapolated to the
population? i.e., Is the sample representative of the population, or has
the population changed?</em></li>
</ul>
<p>Such questions are answered using specific hypothesis tests designed
to deal with such univariate data-based problems.</p>
<p>Example Dataframe:</p>
<pre class="r"><code>set.seed(42)  # For reproducibility

# Generate numerical data
sample_data_large &lt;- rnorm(50, mean = 100, sd = 15)  # Sample size &gt; 30
sample_data_small &lt;- rnorm(20, mean = 100, sd = 15)  # Sample size &lt; 30

# Known population parameters
population_mean &lt;- 100
population_sd &lt;- 15

# Generate categorical data
category_data &lt;- sample(c(&quot;A&quot;, &quot;B&quot;, &quot;C&quot;), 100, replace = TRUE)
ordinal_data &lt;- sample(c(&quot;Low&quot;, &quot;Medium&quot;, &quot;High&quot;), 100, replace = TRUE)</code></pre>
<ol style="list-style-type: lower-alpha">
<li><strong>Z Test:</strong> Used for numerical (quantitative) data
where the sample size is greater than 30 and the population’s standard
deviation is known.</li>
</ol>
<pre class="r"><code># Z Test: Test if sample mean is significantly different from population mean
library(stats)

# Perform Z Test
z_score &lt;- (mean(sample_data_large) - population_mean) / (population_sd / sqrt(length(sample_data_large)))
z_score</code></pre>
<pre><code>[1] -0.2522376</code></pre>
<pre class="r"><code>p_value_z &lt;- 2 * pnorm(-abs(z_score))  # Two-tailed test
p_value_z</code></pre>
<pre><code>[1] 0.8008574</code></pre>
<p>Interpretation: If the p-value is less than the significance level
(commonly 0.05), the sample mean is significantly different from the
population mean.</p>
<ol start="2" style="list-style-type: lower-alpha">
<li><strong>One-Sample t-Test:</strong> Used for numerical
(quantitative) data where the sample size is less than 30 or the
population’s standard deviation is unknown.</li>
</ol>
<pre class="r"><code># One-Sample t-Test: Test if sample mean is significantly different from population mean
t_test_result &lt;- t.test(sample_data_small, mu = population_mean)
t_test_result</code></pre>
<pre><code>
    One Sample t-test

data:  sample_data_small
t = 1.2497, df = 19, p-value = 0.2266
alternative hypothesis: true mean is not equal to 100
95 percent confidence interval:
  97.17831 111.18375
sample estimates:
mean of x 
  104.181 </code></pre>
<p>Interpretation: The t-test result provides a p-value and confidence
interval for the sample mean. A p-value less than 0.05 indicates a
significant difference from the population mean.</p>
<ol start="3" style="list-style-type: lower-alpha">
<li><strong>Chi-Square Test:</strong> Used with ordinal categorical
data</li>
</ol>
<pre class="r"><code># Chi-Square Test: Test the distribution of categorical data
observed_counts &lt;- table(category_data)
expected_counts &lt;- rep(length(category_data) / length(observed_counts), length(observed_counts))

chi_square_result &lt;- chisq.test(observed_counts, p = expected_counts / sum(expected_counts))
chi_square_result</code></pre>
<pre><code>
    Chi-squared test for given probabilities

data:  observed_counts
X-squared = 2.18, df = 2, p-value = 0.3362</code></pre>
<p>Interpretation: The Chi-Square test assesses whether the observed
frequencies differ from the expected frequencies. A p-value less than
0.05 suggests a significant difference.</p>
<ol start="4" style="list-style-type: lower-alpha">
<li><strong>Kolmogorov-Smirnov Test:</strong> Used with nominal
categorical data</li>
</ol>
<pre class="r"><code># Kolmogorov-Smirnov Test: Compare sample distribution to a normal distribution
ks_test_result &lt;- ks.test(sample_data_large, &quot;pnorm&quot;, mean = population_mean, sd = population_sd)
ks_test_result</code></pre>
<pre><code>
    Exact one-sample Kolmogorov-Smirnov test

data:  sample_data_large
D = 0.077011, p-value = 0.906
alternative hypothesis: two-sided</code></pre>
<p>Interpretation: The KS test assesses whether the sample follows the
specified distribution. A p-value less than 0.05 indicates a significant
deviation from the normal distribution.</p>
</div>
<div id="visualising-univariate-data" class="section level3">
<h3>4. Visualising Univariate Data:</h3>
<p>Visualizing univariate data helps us understand the distribution and
patterns within a single variable. Below, we’ll cover visualization
techniques for both categorical and numeric data.</p>
<p>Example Data:</p>
<pre class="r"><code>set.seed(42)  # For reproducibility

# Numeric data
numeric_data &lt;- rnorm(100, mean = 50, sd = 10)

# Categorical data
categorical_data &lt;- sample(c(&quot;Category A&quot;, &quot;Category B&quot;, &quot;Category C&quot;), 100, replace = TRUE)</code></pre>
</div>
</div>
<div id="hypothesis-testing" class="section level1">
<h1>Hypothesis Testing:</h1>
<div id="t-test" class="section level2">
<h2>1. T-Test:</h2>
<p>A T-test is used to determine if there is a significant difference
between the means of two groups. It is typically used when comparing the
means of two groups to see if they are statistically different from each
other.</p>
<p><em>When to use?</em></p>
<ul>
<li><p>When comparing the means of two independent groups (Independent
T-test).</p></li>
<li><p>When comparing the means of two related groups or paired samples
(Paired T-test).</p></li>
</ul>
<pre class="r"><code># Example Data
method_A &lt;- c(85, 88, 90, 92, 87)
method_B &lt;- c(78, 82, 80, 85, 79)

# Perform independent T-test
t_test_result &lt;- t.test(method_A, method_B)

# Print results
print(t_test_result)</code></pre>
<pre><code>
    Welch Two Sample t-test

data:  method_A and method_B
t = 4.3879, df = 7.9943, p-value = 0.002328
alternative hypothesis: true difference in means is not equal to 0
95 percent confidence interval:
  3.605389 11.594611
sample estimates:
mean of x mean of y 
     88.4      80.8 </code></pre>
<pre class="r"><code># Perform independent T-test
t_test_result &lt;- t.test(method_A, method_B, paired = TRUE)

# Print results
print(t_test_result)</code></pre>
<pre><code>
    Paired t-test

data:  method_A and method_B
t = 11.206, df = 4, p-value = 0.0003612
alternative hypothesis: true mean difference is not equal to 0
95 percent confidence interval:
 5.716923 9.483077
sample estimates:
mean difference 
            7.6 </code></pre>
<p><em>Interpretation:</em> <code>p-value &lt; 0.05</code> = there is a
significant difference between the paired samples.</p>
</div>
<div id="wilcoxon-test" class="section level2">
<h2>2. Wilcoxon Test:</h2>
<p>Mann–Whitney U test (also called the Mann–Whitney–Wilcoxon (MWW/MWU),
Wilcoxon rank-sum test, or Wilcoxon–Mann–Whitney test) is a
nonparametric statistical test to compare two independent groups.</p>
<p>The Wilcoxon Signed-Rank Test is a non-parametric test used to
compare two related samples or paired observations to determine if their
population mean ranks differ.</p>
<p><em>When to use?</em></p>
<ul>
<li>When the data is paired and does not meet the assumptions required
for a T-test (e.g., non-normality).</li>
</ul>
<pre class="r"><code># Example Data
before &lt;- c(5, 7, 8, 6, 9)
after &lt;- c(6, 8, 7, 7, 10)

# Perform Wilcoxon rank-sum test Test
wilcox_test_result &lt;- wilcox.test(before, after)</code></pre>
<pre><code>Warning in wilcox.test.default(before, after): cannot compute exact p-value
with ties</code></pre>
<pre class="r"><code># Print results
print(wilcox_test_result)</code></pre>
<pre><code>
    Wilcoxon rank sum test with continuity correction

data:  before and after
W = 10, p-value = 0.6704
alternative hypothesis: true location shift is not equal to 0</code></pre>
<pre class="r"><code># Perform Wilcoxon Signed-Rank Test
wilcox_test_result &lt;- wilcox.test(before, after, paired = TRUE)</code></pre>
<pre><code>Warning in wilcox.test.default(before, after, paired = TRUE): cannot compute
exact p-value with ties</code></pre>
<pre class="r"><code># Print results
print(wilcox_test_result)</code></pre>
<pre><code>
    Wilcoxon signed rank test with continuity correction

data:  before and after
V = 3, p-value = 0.233
alternative hypothesis: true location shift is not equal to 0</code></pre>
<p><em>Interpretation:</em> <code>p-value &lt; 0.05</code> = there is a
significant difference between the paired samples.</p>
</div>
<div id="shapiro-wilk-test-for-normality" class="section level2">
<h2>3. Shapiro-Wilk Test for Normality:</h2>
<p>The Shapiro-Wilk test assesses whether a sample comes from a normally
distributed population. It is particularly useful for checking the
normality assumption in parametric tests like the T-test and ANOVA.</p>
<p><em>When to use?</em></p>
<ul>
<li><p>When you need to check if your data is normally distributed
before performing parametric tests.</p></li>
<li><p>To validate the assumptions of normality for statistical tests
that assume data is normally distributed.</p></li>
</ul>
<pre class="r"><code># Example Data
sample_data &lt;- c(5.2, 6.1, 5.8, 7.2, 6.5, 5.9, 6.8, 6.0, 6.7, 5.7)

# Perform Shapiro-Wilk test
shapiro_test_result &lt;- shapiro.test(sample_data)

# Print results
print(shapiro_test_result)</code></pre>
<pre><code>
    Shapiro-Wilk normality test

data:  sample_data
W = 0.97508, p-value = 0.9335</code></pre>
<p><em>Interpretation:</em> The Shapiro-Wilk test returns a p-value that
indicates whether the sample deviates from a normal distribution.</p>
<ul>
<li><p>p-value &gt; 0.05: Fail to reject the null hypothesis; data is
not significantly different from a normal distribution.</p></li>
<li><p>p-value ≤ 0.05: Reject the null hypothesis; data significantly
deviates from a normal distribution.</p></li>
</ul>
</div>
<div id="multiple-comparisons" class="section level2">
<h2>4. Multiple comparisons:</h2>
<p>ANOVA and Kruskal-Wallis Rank Sum Test are used to determine if there
are any statistically significant differences between the means of three
or more independent groups.</p>
<p><em>When to use?</em></p>
<ul>
<li>When comparing means among three or more groups.</li>
</ul>
<pre class="r"><code># Example Data
scores &lt;- data.frame(
  score = c(85, 88, 90, 92, 87, 78, 82, 80, 85, 79, 95, 97, 92, 91, 96),
  method = factor(rep(c(&quot;A&quot;, &quot;B&quot;, &quot;C&quot;), each = 5))
)

# Perform ANOVA
anova_result &lt;- aov(score ~ method, data = scores)

# Print summary of results
summary(anova_result)</code></pre>
<pre><code>            Df Sum Sq Mean Sq F value   Pr(&gt;F)    
method       2  451.6  225.80   31.22 1.76e-05 ***
Residuals   12   86.8    7.23                     
---
Signif. codes:  0 &#39;***&#39; 0.001 &#39;**&#39; 0.01 &#39;*&#39; 0.05 &#39;.&#39; 0.1 &#39; &#39; 1</code></pre>
<pre class="r"><code># Perform Kruskal-Wallis Rank Sum Test
kruskal.test(score ~ method, data = scores)</code></pre>
<pre><code>
    Kruskal-Wallis rank sum test

data:  score by method
Kruskal-Wallis chi-squared = 11.576, df = 2, p-value = 0.003064</code></pre>
<p><em>Interpretation:</em> <code>p-value &lt; 0.05</code> = there is a
significant difference between the group means.</p>
<ul>
<li>Post-hoc tests (e.g., Tukey’s HSD) can be used to determine which
specific groups differ.</li>
</ul>
</div>
<div id="two-way-anova" class="section level2">
<h2>5. two-way ANOVA:</h2>
<p>A two-way ANOVA (Analysis of Variance) is used when you want to
examine the effect of two independent variables (also called factors) on
a dependent variable, and to determine whether there is an interaction
between these two factors. It is typically used in situations where:</p>
<p>Two independent variables: You have two categorical variables
(factors) and one continuous dependent variable. For example, you might
examine how both “teaching method” (Factor 1) and “gender” (Factor 2)
affect student test scores (dependent variable).</p>
<p>Main effects: It tests for the individual (main) effects of each
independent variable on the dependent variable. For instance, the main
effect of “teaching method” and the main effect of “gender” on test
scores.</p>
<p>Interaction effect: It examines whether the effect of one independent
variable depends on the level of the other variable. For example, if the
effect of a teaching method on test scores differs based on gender.</p>
<p>Two-way ANOVA is commonly used in experiments where you need to
analyze the effect of two factors and their potential interaction on an
outcome. It’s more efficient than performing two separate one-way ANOVAs
for each factor.</p>
<pre class="r"><code># Example Data
head(npk)</code></pre>
<pre><code>  block N P K yield
1     1 0 1 1  49.5
2     1 1 1 0  62.8
3     1 0 0 0  46.8
4     1 1 0 1  57.0
5     2 1 0 0  59.8
6     2 1 1 1  58.5</code></pre>
<pre class="r"><code># Perform two-way ANOVA
npk.aov &lt;- aov(yield ~  N*K, npk)
summary(npk.aov)</code></pre>
<pre><code>            Df Sum Sq Mean Sq F value Pr(&gt;F)  
N            1  189.3  189.28   6.775 0.0170 *
K            1   95.2   95.20   3.408 0.0797 .
N:K          1   33.1   33.14   1.186 0.2891  
Residuals   20  558.7   27.94                 
---
Signif. codes:  0 &#39;***&#39; 0.001 &#39;**&#39; 0.01 &#39;*&#39; 0.05 &#39;.&#39; 0.1 &#39; &#39; 1</code></pre>
<p><em>Interpretation:</em> <code>p-value &lt; 0.05</code> = there is a
significant difference between the group means.</p>
<ul>
<li>Post-hoc tests (e.g., Tukey’s HSD) can be used to determine which
specific groups differ.</li>
</ul>
</div>
<div id="fisher-test" class="section level2">
<h2>6. Fisher Test:</h2>
<p>The Fisher test is used to determine if there is a significant
association between two categorical variables.</p>
<p><em>When to use?</em></p>
<ul>
<li>When testing the independence of two categorical variables in a
contingency table.</li>
</ul>
<pre class="r"><code># Example Data
study_method &lt;- matrix(c(20, 15, 30, 25), nrow = 2, byrow = TRUE)
rownames(study_method) &lt;- c(&quot;Passed&quot;, &quot;Failed&quot;)
colnames(study_method) &lt;- c(&quot;Method A&quot;, &quot;Method B&quot;)

# Perform Fisher test
fisher_result &lt;- fisher.test(study_method)

# Print results
print(fisher_result)</code></pre>
<pre><code>
    Fisher&#39;s Exact Test for Count Data

data:  study_method
p-value = 0.8312
alternative hypothesis: true odds ratio is not equal to 1
95 percent confidence interval:
 0.4353737 2.8593717
sample estimates:
odds ratio 
  1.109811 </code></pre>
<p><em>Interpretation:</em> <code>p-value &lt; 0.05</code> there is a
significant association between the study method and the passing
rate.</p>
</div>
<div id="correlation" class="section level2">
<h2>7. Correlation:</h2>
<p>Correlation analysis is used to investigate the association between
two or more variables.</p>
<div id="step-1-choose-a-correlation-method" class="section level3">
<h3>Step 1: Choose a Correlation Method</h3>
<p><strong><em>Pearson Correlation</em></strong> measures the linear
relationship between two continuous variables. It assumes both variables
follow a normal distribution.</p>
<p><strong><em>Spearman and Kendall Correlation</em></strong> are
non-parametric and measure the strength and direction of the association
between two ranked variables.</p>
</div>
<div id="step-2-calculate-correlation" class="section level3">
<h3>Step 2: Calculate Correlation</h3>
<ol style="list-style-type: lower-roman">
<li>Pearson Correlation</li>
</ol>
<pre class="r"><code># Pearson correlation test
pearson_res &lt;- cor.test(mtcars$mpg, mtcars$wt, method = &quot;pearson&quot;)
pearson_res</code></pre>
<pre><code>
    Pearson&#39;s product-moment correlation

data:  mtcars$mpg and mtcars$wt
t = -9.559, df = 30, p-value = 1.294e-10
alternative hypothesis: true correlation is not equal to 0
95 percent confidence interval:
 -0.9338264 -0.7440872
sample estimates:
       cor 
-0.8676594 </code></pre>
<ol start="2" style="list-style-type: lower-roman">
<li>Spearman and Kendall Correlation</li>
</ol>
<pre class="r"><code># Spearman correlation test
spearman_res &lt;- cor.test(mtcars$mpg, mtcars$wt, method = &quot;spearman&quot;)
spearman_res</code></pre>
<pre><code>
    Spearman&#39;s rank correlation rho

data:  mtcars$mpg and mtcars$wt
S = 10292, p-value = 1.488e-11
alternative hypothesis: true rho is not equal to 0
sample estimates:
      rho 
-0.886422 </code></pre>
<pre class="r"><code># Kendall correlation test
kendall_res &lt;- cor.test(mtcars$mpg, mtcars$wt, method = &quot;kendall&quot;)
kendall_res</code></pre>
<pre><code>
    Kendall&#39;s rank correlation tau

data:  mtcars$mpg and mtcars$wt
z = -5.7981, p-value = 6.706e-09
alternative hypothesis: true tau is not equal to 0
sample estimates:
       tau 
-0.7278321 </code></pre>
</div>
<div id="step-3-interpretation" class="section level3">
<h3>Step 3: Interpretation</h3>
<p>Correlation Coefficient:</p>
<ul>
<li><p><code>-1</code>: Strong negative correlation (as one variable
increases, the other decreases).</p></li>
<li><p><code>0</code>: No correlation.</p></li>
<li><p><code>1</code>: Strong positive correlation (both variables
increase together).</p></li>
</ul>
<p>P-Value:</p>
<ul>
<li><code>p-value &lt; 0.05</code> indicates a statistically significant
correlation.</li>
</ul>
<p><span
style="color:red;"><strong><em>Exercise:</em></strong></span></p>
<p>Perform a correlation analysis using the <code>mpg</code> and
<code>qsec</code> variables from the <code>mtcars</code> to investigate
the extent of correlation between the two variables. Provide an
interpretation of the correlation coefficient and its p-value.</p>
<details>
<summary>
<em>Example interpretation</em>:
</summary>
<ul>
<li><p>The Pearson correlation coefficient is -0.8677, which points to a
strong negative linear relationship between the variables.</p></li>
<li><p>The p-value is significantly low (p &lt; 0.001), indicating that
the correlation is statistically significant.</p></li>
<li><p>The 95% confidence interval suggests that the true correlation
lies between -0.9338 and -0.7441.</p></li>
</ul>
</div>
<div id="simple-linear-regression" class="section level3">
<h3>3. Simple Linear Regression:</h3>
<pre class="r"><code>x &lt;- mtcars$mpg
y &lt;- mtcars$wt

model = lm(y ~ x)
summary(model)</code></pre>
<pre><code>
Call:
lm(formula = y ~ x)

Residuals:
    Min      1Q  Median      3Q     Max 
-0.6516 -0.3490 -0.1381  0.3190  1.3684 

Coefficients:
            Estimate Std. Error t value Pr(&gt;|t|)    
(Intercept)  6.04726    0.30869  19.590  &lt; 2e-16 ***
x           -0.14086    0.01474  -9.559 1.29e-10 ***
---
Signif. codes:  0 &#39;***&#39; 0.001 &#39;**&#39; 0.01 &#39;*&#39; 0.05 &#39;.&#39; 0.1 &#39; &#39; 1

Residual standard error: 0.4945 on 30 degrees of freedom
Multiple R-squared:  0.7528,    Adjusted R-squared:  0.7446 
F-statistic: 91.38 on 1 and 30 DF,  p-value: 1.294e-10</code></pre>
<pre class="r"><code>par(mfrow = c(2, 2))
plot(model)</code></pre>
<p><img src="figure/Statistics.Rmd/unnamed-chunk-38-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-38-1">
Past versions of unnamed-chunk-38-1.png
</button>
</p>
<div id="fig-unnamed-chunk-38-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/tkcaccia/Tutorials/blob/6301d0ade76fe73d7038cd24aecc9f9f9df2794e/docs/figure/Statistics.Rmd/unnamed-chunk-38-1.png" target="_blank">6301d0a</a>
</td>
<td>
tkcaccia
</td>
<td>
2024-09-16
</td>
</tr>
</tbody>
</table>
</div>
</div>
<div id="checking-assumptions" class="section level4">
<h4><em>Checking Assumptions</em>:</h4>
<div
id="assumption-1-linearity-check-if-the-relationship-between-variables-is-linear."
class="section level5">
<h5><strong>Assumption 1: Linearity</strong> –&gt; Check if the
relationship between variables is linear.</h5>
<p>Plot of <code>x</code> vs <code>y</code>: This scatter plot displays
the relationship between the predictor <code>x</code> and the response
<code>y</code>.</p>
<p>Abline (Regression Line): The <code>abline(model)</code> adds the
fitted regression line to the plot.</p>
<pre class="r"><code>plot(x, y)
abline(model)</code></pre>
<p><img src="figure/Statistics.Rmd/unnamed-chunk-39-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-39-1">
Past versions of unnamed-chunk-39-1.png
</button>
</p>
<div id="fig-unnamed-chunk-39-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/tkcaccia/Tutorials/blob/6301d0ade76fe73d7038cd24aecc9f9f9df2794e/docs/figure/Statistics.Rmd/unnamed-chunk-39-1.png" target="_blank">6301d0a</a>
</td>
<td>
tkcaccia
</td>
<td>
2024-09-16
</td>
</tr>
<tr>
<td>
<a href="https://github.com/tkcaccia/Tutorials/blob/5033c128d8bca78746899004bd82e65b109bffd4/docs/figure/Statistics.Rmd/unnamed-chunk-39-1.png" target="_blank">5033c12</a>
</td>
<td>
tkcaccia
</td>
<td>
2024-09-16
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p><em>What to Look For</em>:</p>
<p><strong>Linear Relationship</strong>: The data points should roughly
form a straight line if the linearity assumption is satisfied. The
fitted regression line should capture the trend of the data points
well.</p>
<p><strong>Non-Linearity</strong>: If the data points show a clear
curvature or systematic pattern not captured by the straight line, this
suggests that the linearity assumption is violated. In such cases,
consider polynomial regression or other non-linear models.</p>
</div>
</div>
</div>
</div>
<div id="performing-the-statistics-easy" class="section level2">
<h2>8. Performing the statistics easy:</h2>
<p>Use the following commands to install KODAMA from GitHub using the
devtools package Once installed, load the KODAMA package into your R
session.</p>
<pre class="r"><code>if (!require(devtools)) {
  install.packages(&quot;devtools&quot;)
}
devtools::install_github(&quot;tkcaccia/KODAMA&quot;)
library(KODAMA)</code></pre>
<p>A short example for the analysis of metadata</p>
<pre class="r"><code>data(clinical)

hosp=clinical[,&quot;Hospital&quot;]
gender=clinical[,&quot;Gender&quot;]
GS=clinical[,&quot;Gleason score&quot;]
BMI=clinical[,&quot;BMI&quot;]
age=clinical[,&quot;Age&quot;]

A=categorical.test(&quot;Gender&quot;,gender,hosp)
B=categorical.test(&quot;Gleason score&quot;,GS,hosp)

C=continuous.test(&quot;BMI&quot;,BMI,hosp,digits=2)
D=continuous.test(&quot;Age&quot;,age,hosp,digits=1)

da=rbind(A,B,C,D)
da</code></pre>
<pre><code>              Feature Hospital A     Hospital B  p-value
v              Gender                                   
          Male, n (%) 81 (100.0)     24 (100.0)         
v1      Gleason score                           1.29e-01
X          3+3, n (%)  30 (37.0)       7 (29.2)         
X.1        3+4, n (%)  30 (37.0)       5 (20.8)         
X.2        4+3, n (%)    8 (9.9)       6 (25.0)         
X.3        4+4, n (%)  13 (16.0)       6 (25.0)         
1   BMI, median [IQR] 32 [25 35]  26 [25 30.25] 9.66e-03
11  Age, median [IQR] 63 [56 74] 74 [57.5 79.5] 3.07e-01</code></pre>
<pre class="r"><code>write.csv(da,&quot;output/Table.csv&quot;)</code></pre>
<p>When you have to repeat the same test several times…</p>
<pre class="r"><code>multi_analysis(clinical[,c(&quot;BMI&quot;,&quot;Age&quot;)],clinical[,&quot;Hospital&quot;],FUN=&quot;continuous.test&quot;)</code></pre>
<pre><code>            Feature Hospital A     Hospital B  p-value      FDR
1 BMI, median [IQR] 32 [25 35]  26 [25 30.25] 9.66e-03 1.93e-02
2 Age, median [IQR] 63 [56 74] 74 [57.5 79.5] 3.07e-01 3.07e-01</code></pre>
<br>
<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-sessioninfo" data-toggle="collapse" data-target="#workflowr-sessioninfo" style="display: block;">
<span class="glyphicon glyphicon-wrench" aria-hidden="true"></span>
Session information
</button>
</p>
<div id="workflowr-sessioninfo" class="collapse">
<pre class="r"><code>sessionInfo()</code></pre>
<pre><code>R version 4.3.3 (2024-02-29)
Platform: aarch64-apple-darwin20 (64-bit)
Running under: macOS Sonoma 14.5

Matrix products: default
BLAS:   /Library/Frameworks/R.framework/Versions/4.3-arm64/Resources/lib/libRblas.0.dylib 
LAPACK: /Library/Frameworks/R.framework/Versions/4.3-arm64/Resources/lib/libRlapack.dylib;  LAPACK version 3.11.0

locale:
[1] en_US.UTF-8/en_US.UTF-8/en_US.UTF-8/C/en_US.UTF-8/en_US.UTF-8

time zone: America/Bogota
tzcode source: internal

attached base packages:
[1] stats     graphics  grDevices utils     datasets  methods   base     

other attached packages:
 [1] KODAMA_3.1      umap_0.2.10.0   Rtsne_0.17      minerva_1.5.10 
 [5] devtools_2.4.5  usethis_3.0.0   corrplot_0.92   ggpubr_0.6.0   
 [9] dplyr_1.1.4     ggplot2_3.5.1   workflowr_1.7.1

loaded via a namespace (and not attached):
 [1] tidyselect_1.2.1  fastmap_1.2.0     promises_1.3.0    digest_0.6.36    
 [5] mime_0.12         lifecycle_1.0.4   ellipsis_0.3.2    processx_3.8.4   
 [9] magrittr_2.0.3    compiler_4.3.3    rlang_1.1.4       sass_0.4.9       
[13] tools_4.3.3       utf8_1.2.4        yaml_2.3.10       knitr_1.48       
[17] ggsignif_0.6.4    askpass_1.2.0     htmlwidgets_1.6.4 pkgbuild_1.4.4   
[21] curl_5.2.1        reticulate_1.39.0 pkgload_1.4.0     abind_1.4-5      
[25] miniUI_0.1.1.1    withr_3.0.1       purrr_1.0.2       grid_4.3.3       
[29] fansi_1.0.6       urlchecker_1.0.1  git2r_0.33.0      profvis_0.3.8    
[33] xtable_1.8-4      colorspace_2.1-1  scales_1.3.0      cli_3.6.3        
[37] rmarkdown_2.27    generics_0.1.3    remotes_2.5.0     rstudioapi_0.16.0
[41] RSpectra_0.16-2   httr_1.4.7        sessioninfo_1.2.2 cachem_1.1.0     
[45] stringr_1.5.1     parallel_4.3.3    vctrs_0.6.5       Matrix_1.6-5     
[49] jsonlite_1.8.8    carData_3.0-5     car_3.1-2         callr_3.7.6      
[53] rstatix_0.7.2     tidyr_1.3.1       jquerylib_0.1.4   glue_1.7.0       
[57] ps_1.7.7          stringi_1.8.4     gtable_0.3.5      later_1.3.2      
[61] munsell_0.5.1     tibble_3.2.1      pillar_1.9.0      htmltools_0.5.8.1
[65] openssl_2.2.1     R6_2.5.1          rprojroot_2.0.4   evaluate_0.24.0  
[69] shiny_1.9.1       lattice_0.22-6    highr_0.11        png_0.1-8        
[73] backports_1.5.0   memoise_2.0.1     broom_1.0.6       httpuv_1.6.15    
[77] bslib_0.8.0       Rcpp_1.0.13       whisker_0.4.1     xfun_0.46        
[81] fs_1.6.4          getPass_0.2-4     pkgconfig_2.0.3  </code></pre>
</div>
</div>
</div>


<!-- Adjust MathJax settings so that all math formulae are shown using
TeX fonts only; see
https://docs.mathjax.org/en/latest/web/configuration.html. This will make
the presentation more consistent at the cost of the webpage sometimes
taking slightly longer to load. Note that this only works because the
footer is added to webpages before the MathJax javascript. -->
<script type="text/x-mathjax-config">
  MathJax.Hub.Config({
    "HTML-CSS": { availableFonts: ["TeX"] }
  });
</script>




</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.odd').parent('tbody').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- tabsets -->

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open');
  });
});
</script>

<!-- code folding -->

<script>
$(document).ready(function ()  {

    // temporarily add toc-ignore selector to headers for the consistency with Pandoc
    $('.unlisted.unnumbered').addClass('toc-ignore')

    // move toc-ignore selectors from section div to header
    $('div.section.toc-ignore')
        .removeClass('toc-ignore')
        .children('h1,h2,h3,h4,h5').addClass('toc-ignore');

    // establish options
    var options = {
      selectors: "h1,h2,h3",
      theme: "bootstrap3",
      context: '.toc-content',
      hashGenerator: function (text) {
        return text.replace(/[.\\/?&!#<>]/g, '').replace(/\s/g, '_');
      },
      ignoreSelector: ".toc-ignore",
      scrollTo: 0
    };
    options.showAndHide = true;
    options.smoothScroll = true;

    // tocify
    var toc = $("#TOC").tocify(options).data("toc-tocify");
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
