<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />


<meta name="author" content="Stefano Cacciatore" />

<meta name="date" content="2024-09-16" />

<title>Visualisation</title>

<script src="site_libs/header-attrs-2.27/header-attrs.js"></script>
<script src="site_libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/united.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<style>h1 {font-size: 34px;}
       h1.title {font-size: 38px;}
       h2 {font-size: 30px;}
       h3 {font-size: 24px;}
       h4 {font-size: 18px;}
       h5 {font-size: 16px;}
       h6 {font-size: 12px;}
       code {color: inherit; background-color: rgba(0, 0, 0, 0.04);}
       pre:not([class]) { background-color: white }</style>
<script src="site_libs/jqueryui-1.13.2/jquery-ui.min.js"></script>
<link href="site_libs/tocify-1.9.1/jquery.tocify.css" rel="stylesheet" />
<script src="site_libs/tocify-1.9.1/jquery.tocify.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<link href="site_libs/highlightjs-9.12.0/textmate.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>
<link href="site_libs/font-awesome-6.4.2/css/all.min.css" rel="stylesheet" />
<link href="site_libs/font-awesome-6.4.2/css/v4-shims.min.css" rel="stylesheet" />

<link rel="icon" href="https://github.com/workflowr/workflowr-assets/raw/main/img/reproducible.png">
<!-- Add a small amount of space between sections. -->
<style type="text/css">
div.section {
  padding-top: 12px;
}
</style>



<style type="text/css">
  code{white-space: pre-wrap;}
  span.smallcaps{font-variant: small-caps;}
  span.underline{text-decoration: underline;}
  div.column{display: inline-block; vertical-align: top; width: 50%;}
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
  ul.task-list{list-style: none;}
    </style>

<style type="text/css">code{white-space: pre;}</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>









<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
img {
  max-width:100%;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
details > summary > p:only-child {
  display: inline;
}
pre code {
  padding: 0;
}
</style>


<style type="text/css">
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #adb5bd;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script type="text/javascript">
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark the anchor link active (and if it's in a dropdown, also mark that active)
  var dropdown = menuAnchor.closest('li.dropdown');
  if (window.bootstrap) { // Bootstrap 4+
    menuAnchor.addClass('active');
    dropdown.find('> .dropdown-toggle').addClass('active');
  } else { // Bootstrap 3
    menuAnchor.parent().addClass('active');
    dropdown.addClass('active');
  }

  // Navbar adjustments
  var navHeight = $(".navbar").first().height() + 15;
  var style = document.createElement('style');
  var pt = "padding-top: " + navHeight + "px; ";
  var mt = "margin-top: -" + navHeight + "px; ";
  var css = "";
  // offset scroll position for anchor links (for fixed navbar)
  for (var i = 1; i <= 6; i++) {
    css += ".section h" + i + "{ " + pt + mt + "}\n";
  }
  style.innerHTML = "body {" + pt + "padding-bottom: 40px; }\n" + css;
  document.head.appendChild(style);
});
</script>

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before, .tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "\e259";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "\e258";
  font-family: 'Glyphicons Halflings';
  border: none;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
  background-color: transparent;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<!-- code folding -->



<style type="text/css">

#TOC {
  margin: 25px 0px 20px 0px;
}
@media (max-width: 768px) {
#TOC {
  position: relative;
  width: 100%;
}
}

@media print {
.toc-content {
  /* see https://github.com/w3c/csswg-drafts/issues/4434 */
  float: right;
}
}

.toc-content {
  padding-left: 30px;
  padding-right: 40px;
}

div.main-container {
  max-width: 1200px;
}

div.tocify {
  width: 20%;
  max-width: 260px;
  max-height: 85%;
}

@media (min-width: 768px) and (max-width: 991px) {
  div.tocify {
    width: 25%;
  }
}

@media (max-width: 767px) {
  div.tocify {
    width: 100%;
    max-width: none;
  }
}

.tocify ul, .tocify li {
  line-height: 20px;
}

.tocify-subheader .tocify-item {
  font-size: 0.90em;
}

.tocify .list-group-item {
  border-radius: 0px;
}


</style>



</head>

<body>


<div class="container-fluid main-container">


<!-- setup 3col/9col grid for toc_float and main content  -->
<div class="row">
<div class="col-xs-12 col-sm-4 col-md-3">
<div id="TOC" class="tocify">
</div>
</div>

<div class="toc-content col-xs-12 col-sm-8 col-md-9">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-bs-toggle="collapse" data-target="#navbar" data-bs-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">R Tutorial</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">Home</a>
</li>
<li>
  <a href="Install.html">R install</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">
    Basic
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
    <li>
      <a href="Introduction.html">Introduction to R and RStudio</a>
    </li>
    <li>
      <a href="Statistics.html">Statistics</a>
    </li>
    <li>
      <a href="Visualisation.html">Data Visualisation</a>
    </li>
  </ul>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">
    Multivariate analysis
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
    <li>
      <a href="Supervised_learning.html">Supervised Learning</a>
    </li>
    <li>
      <a href="Unsupervised_Learning.html">Unsupervised Learning</a>
    </li>
  </ul>
</li>
<li>
  <a href="license.html">License</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        <li>
  <a href="https://github.com/oliverdesousa/R_Tutorials">
    <span class="fab fa-github"></span>
     
    Source code
  </a>
</li>
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div id="header">



<h1 class="title toc-ignore">Visualisation</h1>
<h4 class="author">Stefano Cacciatore</h4>
<h4 class="date">September 16, 2024</h4>

</div>


<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-report" data-toggle="collapse" data-target="#workflowr-report">
<span class="glyphicon glyphicon-list" aria-hidden="true"></span>
workflowr <span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span>
</button>
</p>
<div id="workflowr-report" class="collapse">
<ul class="nav nav-tabs">
<li class="active">
<a data-toggle="tab" href="#summary">Summary</a>
</li>
<li>
<a data-toggle="tab" href="#checks"> Checks <span
class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span>
</a>
</li>
<li>
<a data-toggle="tab" href="#versions">Past versions</a>
</li>
</ul>
<div class="tab-content">
<div id="summary" class="tab-pane fade in active">
<p>
<strong>Last updated:</strong> 2024-09-16
</p>
<p>
<strong>Checks:</strong> <span
class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> 7
<span class="glyphicon glyphicon-exclamation-sign text-danger"
aria-hidden="true"></span> 0
</p>
<p>
<strong>Knit directory:</strong> <code>Tutorials/</code> <span
class="glyphicon glyphicon-question-sign" aria-hidden="true"
title="This is the local directory in which the code in this file was executed.">
</span>
</p>
<p>
This reproducible <a href="https://rmarkdown.rstudio.com">R Markdown</a>
analysis was created with <a
  href="https://github.com/workflowr/workflowr">workflowr</a> (version
1.7.1). The <em>Checks</em> tab describes the reproducibility checks
that were applied when the results were created. The <em>Past
versions</em> tab lists the development history.
</p>
<hr>
</div>
<div id="checks" class="tab-pane fade">
<div id="workflowr-checks" class="panel-group">
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRMarkdownfilestronguptodate">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>R Markdown file:</strong> up-to-date
</a>
</p>
</div>
<div id="strongRMarkdownfilestronguptodate"
class="panel-collapse collapse">
<div class="panel-body">
<p>Great! Since the R Markdown file has been committed to the Git
repository, you know the exact version of the code that produced these
results.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongEnvironmentstrongempty">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Environment:</strong> empty </a>
</p>
</div>
<div id="strongEnvironmentstrongempty" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! The global environment was empty. Objects defined in the
global environment can affect the analysis in your R Markdown file in
unknown ways. For reproduciblity it’s best to always run the code in an
empty environment.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSeedstrongcodesetseed20240905code">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Seed:</strong>
<code>set.seed(20240905)</code> </a>
</p>
</div>
<div id="strongSeedstrongcodesetseed20240905code"
class="panel-collapse collapse">
<div class="panel-body">
<p>The command <code>set.seed(20240905)</code> was run prior to running
the code in the R Markdown file. Setting a seed ensures that any results
that rely on randomness, e.g. subsampling or permutations, are
reproducible.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSessioninformationstrongrecorded">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Session information:</strong>
recorded </a>
</p>
</div>
<div id="strongSessioninformationstrongrecorded"
class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Recording the operating system, R version, and package
versions is critical for reproducibility.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongCachestrongnone">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Cache:</strong> none </a>
</p>
</div>
<div id="strongCachestrongnone" class="panel-collapse collapse">
<div class="panel-body">
<p>Nice! There were no cached chunks for this analysis, so you can be
confident that you successfully produced the results during this
run.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongFilepathsstrongrelative">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>File paths:</strong> relative </a>
</p>
</div>
<div id="strongFilepathsstrongrelative" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Using relative paths to the files within your workflowr
project makes it easier to run your code on other machines.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRepositoryversionstrongahrefhttpsgithubcomtkcacciaTutorialstree8b0b07aea50228ab49a3660e7b1327cb440a37c1targetblank8b0b07aa">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Repository version:</strong>
<a href="https://github.com/tkcaccia/Tutorials/tree/8b0b07aea50228ab49a3660e7b1327cb440a37c1" target="_blank">8b0b07a</a>
</a>
</p>
</div>
<div
id="strongRepositoryversionstrongahrefhttpsgithubcomtkcacciaTutorialstree8b0b07aea50228ab49a3660e7b1327cb440a37c1targetblank8b0b07aa"
class="panel-collapse collapse">
<div class="panel-body">
<p>
Great! You are using Git for version control. Tracking code development
and connecting the code version to the results is critical for
reproducibility.
</p>
<p>
The results in this page were generated with repository version
<a href="https://github.com/tkcaccia/Tutorials/tree/8b0b07aea50228ab49a3660e7b1327cb440a37c1" target="_blank">8b0b07a</a>.
See the <em>Past versions</em> tab to see a history of the changes made
to the R Markdown and HTML files.
</p>
<p>
Note that you need to be careful to ensure that all relevant files for
the analysis have been committed to Git prior to generating the results
(you can use <code>wflow_publish</code> or
<code>wflow_git_commit</code>). workflowr only checks the R Markdown
file, but you know if there are other scripts or data files that it
depends on. Below is the status of the Git repository when the results
were generated:
</p>
<pre><code>
Ignored files:
    Ignored:    .DS_Store
    Ignored:    data/.DS_Store

Untracked files:
    Untracked:  data/COADREAD.clin.merged.picked.txt
    Untracked:  data/COADREAD.rnaseqv2__illuminahiseq_rnaseqv2__unc_edu__Level_3__RSEM_genes_normalized__data.data.txt
    Untracked:  data/Rstudio.png
    Untracked:  data/tumor_size_patient_survival.csv
    Untracked:  output/data.csv
    Untracked:  test.R

Unstaged changes:
    Modified:   analysis/.DS_Store
    Deleted:    analysis/Module_1_Basics.Rmd
    Deleted:    analysis/Module_2_Visualisation.Rmd
    Deleted:    analysis/Module_3_Univariate_Analysis.Rmd
    Deleted:    analysis/Module_5_Unsupervised_Learning.Rmd
    Deleted:    analysis/about.Rmd
    Deleted:    analysis/supervised_learning_final.Rmd

</code></pre>
<p>
Note that any generated files, e.g. HTML, png, CSS, etc., are not
included in this status report because it is ok for generated content to
have uncommitted changes.
</p>
</div>
</div>
</div>
</div>
<hr>
</div>
<div id="versions" class="tab-pane fade">

<p>
These are the previous versions of the repository in which changes were
made to the R Markdown (<code>analysis/Visualisation.Rmd</code>) and
HTML (<code>docs/Visualisation.html</code>) files. If you’ve configured
a remote Git repository (see <code>?wflow_git_remote</code>), click on
the hyperlinks in the table below to view the files as they were in that
past version.
</p>
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
File
</th>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
<th>
Message
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/tkcaccia/Tutorials/blob/8b0b07aea50228ab49a3660e7b1327cb440a37c1/analysis/Visualisation.Rmd" target="_blank">8b0b07a</a>
</td>
<td>
tkcaccia
</td>
<td>
2024-09-16
</td>
<td>
Start my new project
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/tkcaccia/Tutorials/6301d0ade76fe73d7038cd24aecc9f9f9df2794e/docs/Visualisation.html" target="_blank">6301d0a</a>
</td>
<td>
tkcaccia
</td>
<td>
2024-09-16
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/tkcaccia/Tutorials/897778af75990c4268c8b31f4717b5182769c81b/docs/Visualisation.html" target="_blank">897778a</a>
</td>
<td>
tkcaccia
</td>
<td>
2024-09-16
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/tkcaccia/Tutorials/1a34734cb3ef1db7012e9458c9d98fcd6625835a/docs/Visualisation.html" target="_blank">1a34734</a>
</td>
<td>
tkcaccia
</td>
<td>
2024-09-16
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/tkcaccia/Tutorials/5033c128d8bca78746899004bd82e65b109bffd4/docs/Visualisation.html" target="_blank">5033c12</a>
</td>
<td>
tkcaccia
</td>
<td>
2024-09-16
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/tkcaccia/Tutorials/blob/e9cf75197fecd2b17c49839ac8157b084819504e/analysis/Visualisation.Rmd" target="_blank">e9cf751</a>
</td>
<td>
tkcaccia
</td>
<td>
2024-09-16
</td>
<td>
Start my new project
</td>
</tr>
</tbody>
</table>
</div>
<hr>
</div>
</div>
</div>
<div id="visualisation" class="section level1">
<h1>Visualisation</h1>
<p>Data visualisation helps to tell stories about the data in a form
that is appealing and easy to understand and draw conclusions. It is
said that “a picture is worth a thousand words”</p>
<p>R is equipped with powerful predefined visualisation features() and
more add-on features used to create complex but aesthetically pleasing
graphs</p>
<p><strong>Learning Objectives:</strong></p>
<ol style="list-style-type: decimal">
<li><p>Understand and Apply Scatter Plots</p></li>
<li><p>Analyze and Interpret Box Plots</p></li>
<li><p>Create and Customize Heatmaps</p></li>
<li><p>Conduct and Visualize Survival Analysis</p></li>
</ol>
<div id="scatter-plots" class="section level2">
<h2>1. Scatter Plots</h2>
<p>Scatter plots are useful for visualizing the relationship between two
continuous variables.</p>
<pre class="r"><code>library(ggplot2)
library(dplyr)

head(starwars)</code></pre>
<pre><code># A tibble: 6 × 14
  name      height  mass hair_color skin_color eye_color birth_year sex   gender
  &lt;chr&gt;      &lt;int&gt; &lt;dbl&gt; &lt;chr&gt;      &lt;chr&gt;      &lt;chr&gt;          &lt;dbl&gt; &lt;chr&gt; &lt;chr&gt; 
1 Luke Sky…    172    77 blond      fair       blue            19   male  mascu…
2 C-3PO        167    75 &lt;NA&gt;       gold       yellow         112   none  mascu…
3 R2-D2         96    32 &lt;NA&gt;       white, bl… red             33   none  mascu…
4 Darth Va…    202   136 none       white      yellow          41.9 male  mascu…
5 Leia Org…    150    49 brown      light      brown           19   fema… femin…
6 Owen Lars    178   120 brown, gr… light      blue            52   male  mascu…
# ℹ 5 more variables: homeworld &lt;chr&gt;, species &lt;chr&gt;, films &lt;list&gt;,
#   vehicles &lt;list&gt;, starships &lt;list&gt;</code></pre>
<pre class="r"><code>#summary(starwars)

# cleaning the dataset to remove NA values for height &amp; mass 
starwars_QC &lt;- starwars %&gt;%
  filter(!is.na(height)&amp;!is.na(mass))

# filtering the dataset to remove outliers
starwars_QC1 &lt;- starwars_QC %&gt;%
  filter(height &lt; 300 &amp; mass &lt; 500)

# Create scatter plot
ggplot(starwars_QC1, aes(x = height, y = mass)) +
  geom_point() +
  labs(title = &quot;Relationship between Mass and Height&quot;, x = &quot;Height(cm)&quot;, y = &quot;Mass(kg)&quot;) +
  theme_minimal()</code></pre>
<p><img src="figure/Visualisation.Rmd/scatter-plot-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-scatter-plot-1">
Past versions of scatter-plot-1.png
</button>
</p>
<div id="fig-scatter-plot-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/tkcaccia/Tutorials/blob/5033c128d8bca78746899004bd82e65b109bffd4/docs/figure/Visualisation.Rmd/scatter-plot-1.png" target="_blank">5033c12</a>
</td>
<td>
tkcaccia
</td>
<td>
2024-09-16
</td>
</tr>
</tbody>
</table>
</div>
</div>
<div id="correlation-coefficient." class="section level4">
<h4>Correlation coefficient.</h4>
<p>The correlation between 2 variables is found with the
<code>cor()</code> function.</p>
<p>Suppose we want to compute the correlation between height and mass in
the <code>starwars_QC1</code> dataset:</p>
<pre class="r"><code># Calculate correlation
correlation &lt;- cor(starwars_QC1$height, starwars_QC1$mass, use = &quot;complete.obs&quot;)
print(paste(&quot;Correlation coefficient between height and mass:&quot;, round(correlation, 2)))</code></pre>
<pre><code>[1] &quot;Correlation coefficient between height and mass: 0.75&quot;</code></pre>
<p>The <code>Pearson</code> correlation is computed by default with the
cor() function. If you want to compute the <code>Spearman</code> or
<code>Kendall's tau-b</code> correlation, add the argument
<code>method = "spearman"</code></p>
<pre class="r"><code># Fit linear model
model &lt;- lm(mass ~ height, data = starwars_QC1)

# Print model summary
summary(model)</code></pre>
<pre><code>
Call:
lm(formula = mass ~ height, data = starwars_QC1)

Residuals:
    Min      1Q  Median      3Q     Max 
-39.006  -7.804   0.508   4.007  57.901 

Coefficients:
             Estimate Std. Error t value Pr(&gt;|t|)    
(Intercept) -31.25047   12.81488  -2.439   0.0179 *  
height        0.61273    0.07202   8.508 1.14e-11 ***
---
Signif. codes:  0 &#39;***&#39; 0.001 &#39;**&#39; 0.01 &#39;*&#39; 0.05 &#39;.&#39; 0.1 &#39; &#39; 1

Residual standard error: 19.49 on 56 degrees of freedom
Multiple R-squared:  0.5638,    Adjusted R-squared:  0.556 
F-statistic: 72.38 on 1 and 56 DF,  p-value: 1.138e-11</code></pre>
<p>Plotting the regression line for the relationship</p>
<pre class="r"><code>ggplot(starwars_QC1, aes(x = height, y = mass)) +
  geom_point() +
  geom_smooth(method = &quot;lm&quot;, color = &quot;blue&quot;, se = FALSE) +
  labs(title = &quot;Relationship between Height and Mass&quot;,
       x = &quot;Height (cm)&quot;,
       y = &quot;Mass (kg)&quot;) +
  theme_minimal()</code></pre>
<pre><code>`geom_smooth()` using formula = &#39;y ~ x&#39;</code></pre>
<p><img src="figure/Visualisation.Rmd/regression_plot-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-regression_plot-1">
Past versions of regression_plot-1.png
</button>
</p>
<div id="fig-regression_plot-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/tkcaccia/Tutorials/blob/5033c128d8bca78746899004bd82e65b109bffd4/docs/figure/Visualisation.Rmd/regression_plot-1.png" target="_blank">5033c12</a>
</td>
<td>
tkcaccia
</td>
<td>
2024-09-16
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code># Regression diagnostics
par(mfrow = c(2, 2))  # Set up plotting area for 4 plots
plot(model)</code></pre>
<p><img src="figure/Visualisation.Rmd/regresion_diagnostics-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-regresion_diagnostics-1">
Past versions of regresion_diagnostics-1.png
</button>
</p>
<div id="fig-regresion_diagnostics-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/tkcaccia/Tutorials/blob/5033c128d8bca78746899004bd82e65b109bffd4/docs/figure/Visualisation.Rmd/regresion_diagnostics-1.png" target="_blank">5033c12</a>
</td>
<td>
tkcaccia
</td>
<td>
2024-09-16
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
</div>
<div id="box-plots" class="section level2">
<h2>2. Box plots</h2>
<p>A boxplot or a box-and-whisker plot is a standard method used to
display the distribution of a dataset based on its five-number summary
of data points; <code>minimum</code>, <code>first quartile[Q1]</code>,
<code>median</code>, <code>third quartile[Q3]</code>, and
<code>maximum</code>.</p>
<p>A box plot can tell you that your data is symmetrical, has outliers
or skewed</p>
<pre class="r"><code>data(&quot;iris&quot;)
head(iris)</code></pre>
<pre><code>  Sepal.Length Sepal.Width Petal.Length Petal.Width Species
1          5.1         3.5          1.4         0.2  setosa
2          4.9         3.0          1.4         0.2  setosa
3          4.7         3.2          1.3         0.2  setosa
4          4.6         3.1          1.5         0.2  setosa
5          5.0         3.6          1.4         0.2  setosa
6          5.4         3.9          1.7         0.4  setosa</code></pre>
<p>Create a box plot of Sepal Length by Species using the iris
dataset</p>
<pre class="r"><code>ggplot(iris, aes(x = Species, y = Sepal.Length, fill = Species)) +
  geom_boxplot() +
  labs(title = &quot;Box Plot of Sepal Length by Species&quot;,
       x = &quot;Species&quot;,
       y = &quot;Sepal Length (cm)&quot;) +
  theme_minimal()</code></pre>
<p><img src="figure/Visualisation.Rmd/simple_boxplot-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-simple_boxplot-1">
Past versions of simple_boxplot-1.png
</button>
</p>
<div id="fig-simple_boxplot-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/tkcaccia/Tutorials/blob/5033c128d8bca78746899004bd82e65b109bffd4/docs/figure/Visualisation.Rmd/simple_boxplot-1.png" target="_blank">5033c12</a>
</td>
<td>
tkcaccia
</td>
<td>
2024-09-16
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p>We can compare the species list</p>
<pre class="r"><code>#install.packages(&quot;ggpubr&quot;)
library(ggpubr)
library(ggplot2)

# Define the colors for the boxplot
cols &lt;- c(&quot;red&quot;, &quot;blue&quot;, &quot;green&quot;)


# Prepare the dataset (using Sepal.Length as the continuous variable)
df &lt;- data.frame(Sepal.Length = iris$Sepal.Length, Species = iris$Species)

# Define comparisons for pairwise Wilcoxon test
my_comparisons &lt;- list(c(&quot;setosa&quot;, &quot;versicolor&quot;), c(&quot;setosa&quot;, &quot;virginica&quot;), c(&quot;versicolor&quot;, &quot;virginica&quot;))

# Create the boxplot
ggboxplot(df, x = &quot;Species&quot;, y = &quot;Sepal.Length&quot;, width = 0.8, palette = cols,
          fill = &quot;Species&quot;, add = &quot;jitter&quot;, ylim = c(4, 9.5),  # Increase ylim to provide space for labels
          add.params = list(size = 0.5, jitter = 0.2, fill = &quot;Species&quot;),
          shape = 21) +
  labs(y= &quot;Sepal Length&quot;, x = &quot;Species&quot;, title = &quot;Box plot sepal length by species&quot;) +
  stat_compare_means(method = &quot;kruskal.test&quot;) +  # Overall comparison using Kruskal-Wallis test
  stat_compare_means(comparisons = my_comparisons, 
                     method = &quot;wilcox.test&quot;, 
                     label = &quot;p.signif&quot;, 
                     label.y = c(8.5, 9.2, 8.7)) +  # Manually adjust the position of significance labels
  theme(legend.position = &quot;none&quot;, plot.margin = unit(c(2, 1, 1, 1), &quot;cm&quot;))</code></pre>
<p><img src="figure/Visualisation.Rmd/ggboxplot-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-ggboxplot-1">
Past versions of ggboxplot-1.png
</button>
</p>
<div id="fig-ggboxplot-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/tkcaccia/Tutorials/blob/5033c128d8bca78746899004bd82e65b109bffd4/docs/figure/Visualisation.Rmd/ggboxplot-1.png" target="_blank">5033c12</a>
</td>
<td>
tkcaccia
</td>
<td>
2024-09-16
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div id="heatmaps" class="section level2">
<h2>3. Heatmaps</h2>
<p>Heat maps or heatmaps allow us to simultaneously visualize clusters
of samples and features.</p>
<p>There are multiple R packages and functions for drawing interactive
and static heatmaps, including:</p>
<ul>
<li><p><code>heatmap()</code> [R base function, stats package]: Draws a
simple heatmap</p></li>
<li><p><code>heatmap.2()</code> [gplots R package]: Draws an enhanced
heatmap compared to the R base function.</p></li>
<li><p><code>pheatmap()</code> [pheatmap R package]: Draws pretty
heatmaps and provides more control to change the appearance of
heatmaps.</p></li>
<li><p><code>d3heatmap()</code> [d3heatmap R package]: Draws an
interactive/clickable heatmap</p></li>
<li><p><code>Heatmap()</code> [ComplexHeatmap R/Bioconductor package]:
Draws, annotates and arranges complex heatmaps (very useful for genomic
data analysis)</p></li>
</ul>
<p>We shall use the <code>mtcars</code> dataset.</p>
<pre class="r"><code>#standardising the data to make variables comparable
df &lt;- scale(mtcars)
# Default plot
heatmap(df, scale = &quot;none&quot;)</code></pre>
<p><img src="figure/Visualisation.Rmd/heatmap-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-heatmap-1">
Past versions of heatmap-1.png
</button>
</p>
<div id="fig-heatmap-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/tkcaccia/Tutorials/blob/5033c128d8bca78746899004bd82e65b109bffd4/docs/figure/Visualisation.Rmd/heatmap-1.png" target="_blank">5033c12</a>
</td>
<td>
tkcaccia
</td>
<td>
2024-09-16
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p>We can customise by using the <code>RColorBrewer</code> color palette
to change the appearance. The arguments <code>RowSideColors</code> and
<code>ColSideColors</code> are used to annotate rows and columns
respectively.</p>
<pre class="r"><code>library(&quot;RColorBrewer&quot;)
col &lt;- colorRampPalette(brewer.pal(10, &quot;RdYlBu&quot;))(256)
heatmap(df, scale = &quot;none&quot;, col =  col, 
        RowSideColors = rep(c(&quot;blue&quot;, &quot;pink&quot;), each = 16),
        ColSideColors = c(rep(&quot;purple&quot;, 5), rep(&quot;orange&quot;, 6)))</code></pre>
<p><img src="figure/Visualisation.Rmd/heatmap_custom-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-heatmap_custom-1">
Past versions of heatmap_custom-1.png
</button>
</p>
<div id="fig-heatmap_custom-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/tkcaccia/Tutorials/blob/5033c128d8bca78746899004bd82e65b109bffd4/docs/figure/Visualisation.Rmd/heatmap_custom-1.png" target="_blank">5033c12</a>
</td>
<td>
tkcaccia
</td>
<td>
2024-09-16
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div id="survival-plots" class="section level2">
<h2>4. Survival plots</h2>
<p>Survival analysis involves a set of statistical approaches used to
investigate the time it takes for an event of interest to occur.</p>
<p>In cancer studies, survival analysis is used for patients survival
time analyses with typical research questions like;</p>
<ul>
<li><p>What is the impact of certain clinical characteristics on
patient’s survival?</p></li>
<li><p>What is the probability that an individual survives after 3
years?</p></li>
<li><p>Are there differences in survival between groups of
patients?</p></li>
</ul>
<pre class="r"><code>#install.packages(&#39;survminer&#39;)
BiocManager::install(&quot;RTCGA.clinical&quot;) # data for examples</code></pre>
<pre><code>&#39;getOption(&quot;repos&quot;)&#39; replaces Bioconductor standard repositories, see
&#39;help(&quot;repositories&quot;, package = &quot;BiocManager&quot;)&#39; for details.
Replacement repositories:
    CRAN: https://cran.rstudio.com/</code></pre>
<pre><code>Bioconductor version 3.18 (BiocManager 1.30.23), R 4.3.3 (2024-02-29)</code></pre>
<pre><code>Old packages: &#39;abind&#39;, &#39;arrow&#39;, &#39;BiocManager&#39;, &#39;boot&#39;, &#39;broom.helpers&#39;,
  &#39;caTools&#39;, &#39;clock&#39;, &#39;corrplot&#39;, &#39;cpp11&#39;, &#39;curl&#39;, &#39;data.table&#39;, &#39;Deriv&#39;,
  &#39;DescTools&#39;, &#39;digest&#39;, &#39;emmeans&#39;, &#39;EnvStats&#39;, &#39;expm&#39;, &#39;fastDummies&#39;, &#39;gert&#39;,
  &#39;ggrepel&#39;, &#39;golem&#39;, &#39;hardhat&#39;, &#39;harmony&#39;, &#39;httr2&#39;, &#39;lpSolve&#39;, &#39;matrixStats&#39;,
  &#39;microbenchmark&#39;, &#39;minqa&#39;, &#39;mvtnorm&#39;, &#39;nlme&#39;, &#39;openssl&#39;, &#39;ps&#39;, &#39;ragg&#39;,
  &#39;RANN&#39;, &#39;RcppArmadillo&#39;, &#39;RcppEigen&#39;, &#39;recipes&#39;, &#39;reticulate&#39;, &#39;rmarkdown&#39;,
  &#39;sf&#39;, &#39;sfsmisc&#39;, &#39;Signac&#39;, &#39;spatstat.explore&#39;, &#39;spatstat.univar&#39;,
  &#39;spatstat.utils&#39;, &#39;spData&#39;, &#39;spdep&#39;, &#39;tinytex&#39;, &#39;waldo&#39;, &#39;wk&#39;, &#39;xfun&#39;</code></pre>
<pre class="r"><code>library(survminer)
library(RTCGA.clinical)</code></pre>
<pre><code>Loading required package: RTCGA</code></pre>
<pre><code>Welcome to the RTCGA (version: 1.32.0). Read more about the project under https://rtcga.github.io/RTCGA/</code></pre>
<pre class="r"><code>survivalTCGA(BRCA.clinical, OV.clinical,
             extract.cols = &quot;admin.disease_code&quot;) -&gt; BRCAOV.survInfo

library(survival)</code></pre>
<pre><code>
Attaching package: &#39;survival&#39;</code></pre>
<pre><code>The following object is masked from &#39;package:survminer&#39;:

    myeloma</code></pre>
<pre class="r"><code>fit &lt;- survfit(Surv(times, patient.vital_status) ~ admin.disease_code,
               data = BRCAOV.survInfo)

# Visualize with survminer
ggsurvplot(fit, data = BRCAOV.survInfo, risk.table = TRUE)</code></pre>
<p><img src="figure/Visualisation.Rmd/survival-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-survival-1">
Past versions of survival-1.png
</button>
</p>
<div id="fig-survival-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/tkcaccia/Tutorials/blob/5033c128d8bca78746899004bd82e65b109bffd4/docs/figure/Visualisation.Rmd/survival-1.png" target="_blank">5033c12</a>
</td>
<td>
tkcaccia
</td>
<td>
2024-09-16
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p>The simple plot above shows an estimate of survival probability
depending on days from cancer diagnostics grouped by cancer types.
Additionally an informative risk set table which shows the number of
patients that were under observation in the specific period of time.</p>
<p>One can create a powerful informative survival plot with such
specification of the parameters</p>
<pre class="r"><code>ggsurvplot(
  fit,                     # survfit object with calculated statistics.
  data = BRCAOV.survInfo,  # data used to fit survival curves. 
  risk.table = TRUE,       # show risk table.
  pval = TRUE,             # show p-value of log-rank test.
  conf.int = TRUE,         # show confidence intervals for point estimates of survival curves.
  xlim = c(0,2000),        # present narrower X axis, but not affect survival estimates.
  break.time.by = 500,     # break X axis in time intervals by 500.
  ggtheme = theme_minimal(), # customize plot and risk table with a theme.
  risk.table.y.text.col = T, # colour risk table text annotations.
  risk.table.y.text = FALSE # show bars instead of names in text annotations in legend of risk table
)</code></pre>
<p><img src="figure/Visualisation.Rmd/survival_plot_customised-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-survival_plot_customised-1">
Past versions of survival_plot_customised-1.png
</button>
</p>
<div id="fig-survival_plot_customised-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/tkcaccia/Tutorials/blob/5033c128d8bca78746899004bd82e65b109bffd4/docs/figure/Visualisation.Rmd/survival_plot_customised-1.png" target="_blank">5033c12</a>
</td>
<td>
tkcaccia
</td>
<td>
2024-09-16
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div id="pie-charts" class="section level2">
<h2>5. Pie Charts:</h2>
<p>A pie chart represents the proportion of each species in the
dataset.</p>
<pre class="r"><code># Load necessary libraries
library(dplyr)
library(ggplot2)

# Create a data frame with counts of Species
x &lt;- as.data.frame(table(iris$Species))

# Create a pie chart using ggplot2
ggplot(x, aes(x = &quot;&quot;, y = Freq, fill = Var1)) +
  geom_bar(stat = &quot;identity&quot;, width = 1) +
  coord_polar(theta = &quot;y&quot;) +
  labs(title = &quot;Pie Chart&quot;) +
  theme_void() + 
  scale_fill_brewer(palette = &quot;Pastel1&quot;) +  
  theme(legend.title = element_blank(),
        plot.title = element_text(hjust = 0.5, face = &quot;bold&quot;))</code></pre>
<p><img src="figure/Visualisation.Rmd/unnamed-chunk-1-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-1-1">
Past versions of unnamed-chunk-1-1.png
</button>
</p>
<div id="fig-unnamed-chunk-1-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/tkcaccia/Tutorials/blob/5033c128d8bca78746899004bd82e65b109bffd4/docs/figure/Visualisation.Rmd/unnamed-chunk-1-1.png" target="_blank">5033c12</a>
</td>
<td>
tkcaccia
</td>
<td>
2024-09-16
</td>
</tr>
</tbody>
</table>
</div>
</div>
<br>
<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-sessioninfo" data-toggle="collapse" data-target="#workflowr-sessioninfo" style="display: block;">
<span class="glyphicon glyphicon-wrench" aria-hidden="true"></span>
Session information
</button>
</p>
<div id="workflowr-sessioninfo" class="collapse">
<pre class="r"><code>sessionInfo()</code></pre>
<pre><code>R version 4.3.3 (2024-02-29)
Platform: aarch64-apple-darwin20 (64-bit)
Running under: macOS Sonoma 14.5

Matrix products: default
BLAS:   /Library/Frameworks/R.framework/Versions/4.3-arm64/Resources/lib/libRblas.0.dylib 
LAPACK: /Library/Frameworks/R.framework/Versions/4.3-arm64/Resources/lib/libRlapack.dylib;  LAPACK version 3.11.0

locale:
[1] en_US.UTF-8/en_US.UTF-8/en_US.UTF-8/C/en_US.UTF-8/en_US.UTF-8

time zone: America/Bogota
tzcode source: internal

attached base packages:
[1] stats     graphics  grDevices utils     datasets  methods   base     

other attached packages:
[1] survival_3.7-0               RTCGA.clinical_20151101.32.0
[3] RTCGA_1.32.0                 survminer_0.4.9             
[5] RColorBrewer_1.1-3           ggpubr_0.6.0                
[7] dplyr_1.1.4                  ggplot2_3.5.1               
[9] workflowr_1.7.1             

loaded via a namespace (and not attached):
 [1] tidyselect_1.2.1    viridisLite_0.4.2   farver_2.1.2       
 [4] viridis_0.6.5       bitops_1.0-8        fastmap_1.2.0      
 [7] RCurl_1.98-1.16     promises_1.3.0      XML_3.99-0.17      
[10] digest_0.6.36       lifecycle_1.0.4     processx_3.8.4     
[13] magrittr_2.0.3      compiler_4.3.3      rlang_1.1.4        
[16] sass_0.4.9          tools_4.3.3         utf8_1.2.4         
[19] yaml_2.3.10         data.table_1.15.4   knitr_1.48         
[22] ggsignif_0.6.4      labeling_0.4.3      xml2_1.3.6         
[25] abind_1.4-5         withr_3.0.1         purrr_1.0.2        
[28] grid_4.3.3          fansi_1.0.6         git2r_0.33.0       
[31] xtable_1.8-4        colorspace_2.1-1    scales_1.3.0       
[34] cli_3.6.3           rmarkdown_2.27      generics_0.1.3     
[37] rstudioapi_0.16.0   km.ci_0.5-6         httr_1.4.7         
[40] commonmark_1.9.1    cachem_1.1.0        stringr_1.5.1      
[43] splines_4.3.3       ggthemes_5.1.0      rvest_1.0.4        
[46] assertthat_0.2.1    BiocManager_1.30.23 survMisc_0.5.6     
[49] vctrs_0.6.5         Matrix_1.6-5        jsonlite_1.8.8     
[52] carData_3.0-5       car_3.1-2           callr_3.7.6        
[55] rstatix_0.7.2       tidyr_1.3.1         jquerylib_0.1.4    
[58] glue_1.7.0          ps_1.7.7            ggtext_0.1.2       
[61] stringi_1.8.4       gtable_0.3.5        later_1.3.2        
[64] munsell_0.5.1       tibble_3.2.1        pillar_1.9.0       
[67] htmltools_0.5.8.1   R6_2.5.1            KMsurv_0.1-5       
[70] rprojroot_2.0.4     evaluate_0.24.0     lattice_0.22-6     
[73] markdown_1.13       highr_0.11          backports_1.5.0    
[76] gridtext_0.1.5      broom_1.0.6         httpuv_1.6.15      
[79] bslib_0.8.0         Rcpp_1.0.13         gridExtra_2.3      
[82] nlme_3.1-165        mgcv_1.9-1          whisker_0.4.1      
[85] xfun_0.46           fs_1.6.4            zoo_1.8-12         
[88] getPass_0.2-4       pkgconfig_2.0.3    </code></pre>
</div>
</div>
</div>


<!-- Adjust MathJax settings so that all math formulae are shown using
TeX fonts only; see
https://docs.mathjax.org/en/latest/web/configuration.html. This will make
the presentation more consistent at the cost of the webpage sometimes
taking slightly longer to load. Note that this only works because the
footer is added to webpages before the MathJax javascript. -->
<script type="text/x-mathjax-config">
  MathJax.Hub.Config({
    "HTML-CSS": { availableFonts: ["TeX"] }
  });
</script>




</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.odd').parent('tbody').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- tabsets -->

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open');
  });
});
</script>

<!-- code folding -->

<script>
$(document).ready(function ()  {

    // temporarily add toc-ignore selector to headers for the consistency with Pandoc
    $('.unlisted.unnumbered').addClass('toc-ignore')

    // move toc-ignore selectors from section div to header
    $('div.section.toc-ignore')
        .removeClass('toc-ignore')
        .children('h1,h2,h3,h4,h5').addClass('toc-ignore');

    // establish options
    var options = {
      selectors: "h1,h2,h3",
      theme: "bootstrap3",
      context: '.toc-content',
      hashGenerator: function (text) {
        return text.replace(/[.\\/?&!#<>]/g, '').replace(/\s/g, '_');
      },
      ignoreSelector: ".toc-ignore",
      scrollTo: 0
    };
    options.showAndHide = true;
    options.smoothScroll = true;

    // tocify
    var toc = $("#TOC").tocify(options).data("toc-tocify");
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
